app/code/community/Businessfactory/Roihuntereasy/Block/CheckoutAnalytics.php

```diff
@@ -4,49 +4,82 @@ class Businessfactory_Roihuntereasy_Block_CheckoutAnalytics extends Businessfact
-            $product_remarketing = json_decode($product_remarketing_json, true);
+            $productRemarketingBase64 = Mage::getSingleton("customer/session")->getMyValue();
+            $productRemarketingJson = base64_decode($productRemarketingBase64);
+            $productRemarketing = json_decode($productRemarketingJson, true);
 
-            if ($product_remarketing && array_key_exists('pagetype', $product_remarketing)) {
-                $pagetype = $product_remarketing['pagetype'];
+            if ($productRemarketing && array_key_exists("pagetype", $productRemarketing)) {
+                $pagetype = $productRemarketing["pagetype"];
 
                 // render template with remarketing tag
-                if ($pagetype === "checkout" && $product_remarketing) {
-                    $this->prodId = $product_remarketing['id'];
-                    $this->prodPrice = $product_remarketing['price'];
+                if ($pagetype === "checkout" && $productRemarketing) {
+                    $this->prodId = $productRemarketing["id"];
+                    $this->prodPrice = $productRemarketing["price"];
+                    $this->conversionCurrency = $productRemarketing["currency"];
 
                     // unset session value
-                    Mage::getSingleton('customer/session')->unsMyValue();
+                    Mage::getSingleton("customer/session")->unsMyValue();
 
                     return parent::_toHtml();
                 }
             }
         } catch (Exception $exception) {
-            Mage::log(__METHOD__ . " exception.", null, 'errors.log');
-            Mage::log($exception, null, 'errors.log');
+            Mage::log(__METHOD__ . " exception.", null, "errors.log");
+            Mage::log($exception->getMessage(), null, "errors.log");
         }
 
-        return '';
+        return "";
     }
 
     public function getProdId()
     {
         if (!$this->prodId) {
-            Mage::log("Product ID not found during " . __METHOD__, null, 'errors.log');
+            Mage::log("Product ID not found during " . __METHOD__, null, "errors.log");
         }
         return json_encode($this->prodId);
     }
 
     public function getProdPrice()
     {
         if (!$this->prodPrice) {
-            Mage::log("Product price not found during " . __METHOD__, null, 'errors.log');
+            Mage::log("Product price not found during " . __METHOD__, null, "errors.log");
         }
         return $this->prodPrice;
     }
+
+    public function getConversionLabel()
+    {
+        try {
+            $collection=$this->getCollection();
+
+            if (($mainItem = ($collection->getLastItem())) == NULL) {
+                Mage::log("Table record not found during " . __METHOD__, null, "errors.log");
+                return null;
+            }
+
+            if (($conversionId = $mainItem->getConversionLabel()) == NULL) {
+                Mage::log("Conversion ID not found during " . __METHOD__, null, "errors.log");
```

Comment:
Please fix property naming and log messages, it's not clear if this function return conversion Id or conversion label.